### ISSUE TITLE
[PRE-RELEASE] Light & Dark Theme - Contrast issues

### ISSUE THREAD

**Issue by OLLI-S** [keepassxreboot/keepassxc#4407]
## Requirements
All text elements, images containing text and relevant elements without text (e.g. icons) a **minimum colour contrast ratio** of **4.5:1** is required (to fulfil **level AA** of the WCAG). For content with a font size of at least 18 Point or 14 Point plus bold a colour contrast ratio of 3.0:1 is required. 
**Disabled elements** (like disabled toolbar icons or disabled edit fields) are **excluded** from measurement.

**International standards:**
- ISO 9241-171 - 10.4.5
- WCAG 2.0 - 1.4.3

I checked the colour contrast in KeePassXC and for most elements the colour contrast is fulfilling the requirements. So here I list only those elements where the colour contrast is not sufficient.
I know there is a gap between design (what it looking good and sexy) and the required colour contrast. But these requirements are important for all users, not only for users that have visual impairments.

## Results for the Light Theme
In the **Light Theme** the following elements do **not fulfil the requirements** described above (the colour contrast ratio is less than 4.5:1 at these elements):

1. Selected item in the menu bar (like "Database") and in the context menu: **3.5:1**
2. Tooltip on toolbar-icons: **3.8:1**
3. Green Button "OK": **3.3:1**
4. Green headlines of group boxes in Search-Help: **3.5:1**
5. Green headlines of group boxes when editing an entry (like "Additional Attributes"): **3.4:1**
6. Arrows to expand/collapse the groups: **4.2:1** (maybe you take the same colour (#686970) that you are using for the arrows that indicate the sort-order in tables, they have 4.7:1)
7. URLs in Entry-Preview-Panel: **3.2:1**
8. Selected entry in lists (like groups, list of entries) wen the list **is focussed** (I clicked on a group and measured the selected group): **3.5:1**
9. Selected entry in lists (like groups, list of entries) wen the list **is not focussed** (I clicked on a group and measured the selected entry in the password list): **3.6:1**
10. Arrow-Icon in the combo box "Username": **3.7:1**
11. *Password Quality "Weak" in the "Password Generator" tool: **2.4:1***
12. *Password Quality "Good" in the "Password Generator" tool: **3.2:1***
13. *Password Quality "Excellent" in the "Password Generator" tool: **4.2:1***
14. *Password Quality "Weak" when editing a password: **2.2:1***
15. *Password Quality "Good" when editing a password: **3.0:1***
16. *Password Quality "Excellent" when editing a password: **3.9:1***
17. Links in the settings (Browser integration) and in the About window: **3.2:1**

## Results for the Dark Theme
In the **Dark Theme** the following elements do **not fulfil the requirements** described above:

18. Arrows to expand/collapse the groups: **3.3:1**
19. Arrows that indicate the sort-order in tables: **2.7:1**
20. URLs in Entry-Preview-Panel: **4.2:1**
21. Arrow-Icon in the combo box "Username": **3.6:1**
22. *Password Quality "Poor" in the "Password Generator" tool: **2.6:1***
23. *Password Quality "Good" in the "Password Generator" tool: **4.2:1***
24. *Password Quality "Excellent" in the "Password Generator" tool: **3.2:1***
25. *Password Quality "Poor" when editing a password: **2.2:1***
26. *Password Quality "Good" when editing a password: **3.5:1****
27. *Password Quality "Excellent" when editing a password: **2.6:1***
28. Links in the settings (Browser integration) and in the About window: **4.2:1**

**Annotations**
* At the numbers 11 to 16 (in the light theme) and 22 to 27 (dark theme) I measured the contrast of the coloured bar to the background (below the bar).

* I would say that the **password quality bar** is an **decorative element** (because users also see the text) so in my opinion there is nothing that should be changed here.
This is the reason, why I wrote these entries in *cursive text*

## Debug Info
KeePassXC - Version 2.5.3-snapshot
Build Type: Snapshot
Revision: 557736e

Qt 5.12.0
Debugging mode is disabled.

Operating system: Windows 10 (10.0)
CPU architecture: x86_64
Kernel: winnt 10.0.18363

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.4

**Comment by tomhundt** [comment]
_Possibly related:_  The contrast for **a "clone" entry** (i.e., using "{Ref:" for username/pw), in **Dark theme**, is really low (i.e., it's black on very dark grey, or on pretty dark grey, depending on which alternating colored line it's on):
<img width="577" alt="expire-entries-dark-theme" src="https://user-images.githubusercontent.com/12475662/76150641-8e142300-6061-11ea-82ee-becc3d620395.png">

I'm running on MacOS 10.15.4 Beta 4 (19E250c),
```
KeePassXC - Version 2.5.3-snapshot
Build Type: Snapshot
Revision: e26063a
```

**Comment by phoerious** [comment]
The reference colour is in fact a different issue, but definitely a bug that needs to be fixed. The colour is hard-coded at the moment, which doesn't work on dark backgrounds.
